components/TimelineFilter.js

```diff
@@ -0,0 +1,56 @@
+import styled from "styled-components";
+
+export default function TimelineFilter({
+  selectedTimeframe,
+  onTimeframeChange,
+}) {
+  const timeframes = [
+    { label: "7 Days", value: 7 },
+    { label: "30 Days", value: 30 },
+    { label: "Half Year", value: 180 },
+    { label: "Full Year", value: 365 },
+  ];
+
+  function handleTimeframeClick(value) {
+    if (selectedTimeframe === value) {
+      onTimeframeChange(null);
+    } else {
+      onTimeframeChange(value);
+    }
+  }
```

Comment:
you could move this function to the index.js and replace the function 'handleTimeFrameChange' with it.

Reply:
Hi,
nice hint, thanks.
i moved it to the index.js and replace it in the TimelineFilter component.